### PR TITLE
Sy 1034 search query attempting to find deleted resources

### PR DESCRIPTION
**Pull request by emilbon99**
# Fix Pull Request Template

## Key Information

- **Linear Issue**: [SY-1034](https://linear.app/synnaxlabs/issue/SY-1034/search-query-attempting-to-find-deleted-resources)

## Description

Fixes an issue where searching the ontology would incorrectly return a not found error even if the resources were found. This fix addresses the issue, and makes ignores `query.NotFound` error so we do a best effort search even if the search index and ontology are slightly out of sync.

Also makes a minor CSS change to the haul drop region css styles (converts them to gray instead of the primary color).

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [x] I have added sufficient regression tests to cover the changes to CI.
- [x] I have added relevant tests to cover the changes or exposing bugs.
- [x] I have verified code coverage targets are met.

## Additional Notes
- [ ] These changes deal with concurrency.
- [ ] These changes affect UI.

## Manual QA Additions

- [ ] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
  with necessary manual QA steps to test my change.

## Reviewer Checklist
- [ ] Sufficient test coverage of new additions.
- [ ] Verified all steps in readiness checklists.
- [ ] UI changes have been tested.
- [ ] style and formatting is consistent.
- [ ] Reviewed any relevant changes to concurrent code for safety. 
- [ ] Sufficient comments and clarity of code.

### CODE REVIEW

File: synnax/pkg/distribution/ontology/retrieve.go [hunk 1]
```diff
@@ -219,29 +218,36 @@ func (r Retrieve) retrieveEntities(
...
-	includeSchema := getIncludeSchema(clause.Params)
-	for j, res := range entries.All() {
+	var (
+		entries          = gorp.GetEntries[ID, Resource](clause.Params)
+		excludeFieldData = getExcludeFieldData(clause.Params)
+		includeSchema    = getIncludeSchema(clause.Params)
+		retrieveResource = (!excludeFieldData) || includeSchema
+	)
+	// Iterate over the entries in place, retrieving the resource if the query requires it.
+	err := entries.MapInPlace(func(res Resource) (Resource, bool, error) {
```
Comment (Lham42) [inline synnax/pkg/distribution/ontology/retrieve.go:228]: what does the bool represent in the return of this function passed in?

**Review by Lham42 — APPROVED** [review]
Approved once you answer my one comment

File: synnax/pkg/distribution/ontology/retrieve.go (hunk 1 above)
Comment (emilbon99) [reply synnax/pkg/distribution/ontology/retrieve.go:228]: Whether the returned item should be included in the mapped result. Essentially a built-in filter.

Comment (emilbon99) [reply synnax/pkg/distribution/ontology/retrieve.go:228]: Also documented in the function docstring :)

**Pull request merged**